extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -249,9 +277,17 @@ private static boolean parquetFileExists(@NotNull final URI fileURI) {
         return !fileURI.getScheme().equals(FILE_URI_SCHEME) || Files.exists(Path.of(fileURI));
     }
 
+    @Override
+    @Nullable
+    public final BasicDataIndex getDataIndex(@NotNull final String... columns) {
+        initialize();
+        return super.getDataIndex(columns);
+    }
```

Comment:
This seems like a poorly scoped `initialize`; it is not clear *what* member variables this is guarding. Assuming the member variables are guarded correctly, anything that `super.getDataIndex` calls should _already_ be properly guarded.